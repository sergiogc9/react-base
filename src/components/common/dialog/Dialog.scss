@import "../../../scss/globals";

.react-base-dialog {
	> div {
		display: flex;
		flex-direction: column;
		width: 77rem;
		height: 90%;
	}

	.dialog-close {
		position: absolute;
		top: 1.5rem;
		right: 1.5rem;

		i {
			color: black;
			font-size: 2.5rem;
		}
	}

	.dialog-header {
		.dialog-logo {
			flex-grow: 0;
			margin-right: 3.5rem;

			i {
				color: $react-base-green;
				font-size: 7rem;
			}

			+ .dialog-title {
				max-width: 40rem;
			}
		}

		.dialog-title {
			max-width: 50rem;
			line-height: 3.5rem;
			font-weight: bold;
			font-size: 3.5rem;
			align-items: center;
			word-break: break-word;
		}
	}

	.md-dialog-content {
		padding: 2.5rem 5%;
		overflow: unset;
		height: 100%;
		display: flex;
		flex-grow: 1;
		max-height: none !important;

		.dialog-container {
			height: 100%;
			overflow-y: auto;
			flex-grow: 1;
			.dialog-content {
				width: 100%;
				color: #757575;
				font-size: 1.4rem;
				padding-bottom: 7rem;
			}

		}
	}
		.dialog-footer {
		justify-content: space-between;
		background-color: #fff;
		display: flex;
		width: 100%;
		flex-direction: row;
		padding: 1rem 5%;

		&.center-buttons {
			justify-content: center;
		}

		button {
			padding: 0.5rem;
			height: 4rem;
			border-radius: 3px;

			&.btn--grey {
				background-color: $md-grey-300;
			}

			&.btn--green {
				background-color: $react-base-green;
			}
		}
	}
}
