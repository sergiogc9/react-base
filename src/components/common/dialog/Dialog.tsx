import React from 'react';
import DialogContainer from 'react-md/lib/Dialogs/DialogContainer';
import FontIcon from 'react-md/lib/FontIcons/FontIcon';
import Button from 'react-md/lib/Buttons/Button';

import { withT } from '@src/lib/i18n';
import { ComponentProps } from './types';

import './Dialog.scss';

const Dialog = (props: ComponentProps) => {

	const { id, icon, title, content, onAccept, onAcceptText, onCancel, onCancelText } = props;

	const centerButtons = (!onAccept && onCancel) || (onAccept && !onCancel) ? "center-buttons" : "";
	const footerClassName = `dialog-footer md-grid ${centerButtons}`;

	const actions = [];
	if (onCancel) actions.push(<Button id="react-baseDialogCancelBtn" className="btn--grey btn--no-background" flat onClick={onCancel}>{onCancelText}</Button>);
	if (onAccept) actions.push(<Button id="react-baseDialogAcceptBtn" className="btn--green btn--no-background" flat onClick={onAccept}>{onAcceptText}</Button>);

	const dialogHeader = (
		<>
			<div className="dialog-close">
				{onCancel ? <Button id="react-baseDialogClosBtn" className="btn--no-background" flat onClick={onCancel}><FontIcon>clear</FontIcon></Button> : null}
			</div>
			<div className="dialog-header md-grid">
				{icon ? <div className="dialog-logo">{icon}</div> : null}
				<div className="dialog-title">{title}</div>
			</div>
		</>
	);

	const dialogContent = (
		<div className="dialog-container">
			<div className="dialog-content">{content}</div>
		</div>
	);

	return <DialogContainer
		id={id}
		className="react-base-dialog"
		visible={true}
		onHide={() => {
			if (onCancel) onCancel();
			else if (onAccept) onAccept();
		}}
		focusOnMount={false}
		aria-labelledby="react-base-dialog"
		title={dialogHeader}
		children={dialogContent}
		footerClassName={footerClassName}
		actions={actions}
	/>;
};

export default withT(Dialog);
