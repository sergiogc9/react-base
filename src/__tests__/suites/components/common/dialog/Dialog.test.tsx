import { mount } from 'enzyme';
import React from 'react';
import FontIcon from 'react-md/lib/FontIcons/FontIcon';

import Dialog from '@src/components/common/dialog';

let onAcceptMock: jest.Mock;
let onCancelMock: jest.Mock;

describe('<Dialog />', () => {
	let wrapper: any;

	beforeEach(() => {
		onAcceptMock = jest.fn();
		onCancelMock = jest.fn();
		wrapper = mount((<Dialog
			id="dialog-id"
			title="Dialog title"
			content={(<div id="dialog-content"></div>)}
			icon={<FontIcon>warning</FontIcon>}
			onAccept={onAcceptMock}
			onAcceptText="Accept"
			onCancel={onCancelMock}
			onCancelText="Cancel"
		></Dialog>));
	});

	it("Test Dialog component snapshot", () => {
		expect(wrapper.html()).toMatchSnapshot();
		expect(wrapper.find('#dialog-content')).toHaveLength(1);
		expect(wrapper.find(FontIcon)).toHaveLength(2); // Close button is also a FontIcon
	});

	it("Test Dialog without icon", () => {
		wrapper.setProps({ icon: undefined });
		expect(wrapper.find(FontIcon)).toHaveLength(1);
	});

	it("Test Dialog without accept button", () => {
		wrapper.setProps({ onAccept: undefined });
		expect(wrapper.find('#react-baseDialogAcceptBtn')).toHaveLength(0);
	});

	it("Test Dialog without cancel button", () => {
		wrapper.setProps({ onCancel: undefined });
		expect(wrapper.find('#react-baseDialogCancelBtn')).toHaveLength(0);
	});

	it("Test Dialog footer centered buttons class", () => {
		wrapper.setProps({ onCancel: undefined }); // onAccept is set
		expect(wrapper.find('footer').hasClass('center-buttons')).toBe(true);
	});
});
