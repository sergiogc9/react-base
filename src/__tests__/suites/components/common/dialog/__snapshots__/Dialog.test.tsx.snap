// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> Test Dialog component snapshot 1`] = `"<span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover launchmetrics-dialog\\" tabindex=\\"-1\\"><div aria-labelledby=\\"launchmetrics-dialog\\" id=\\"dialog-id\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><h2 id=\\"dialog-id-title\\" class=\\"md-title md-title--dialog\\"><div class=\\"dialog-close\\"><button id=\\"launchMetricsDialogClosBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--no-background\\"><div class=\\"md-ink-container\\"></div><i class=\\"md-icon material-icons\\">clear</i></button></div><div class=\\"dialog-header md-grid\\"><div class=\\"dialog-logo\\"><i class=\\"md-icon material-icons\\">warning</i></div><div class=\\"dialog-title\\">Dialog title</div></div></h2><section class=\\"md-dialog-content md-dialog-content--padded\\"><span aria-hidden=\\"true\\"></span><div class=\\"dialog-container\\"><div class=\\"dialog-content\\"><div id=\\"dialog-content\\"></div></div></div></section><footer class=\\"md-dialog-footer md-dialog-footer--stacked dialog-footer md-grid \\"><button id=\\"launchmetricsDialogCancelBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block md-btn--dialog btn--grey btn--no-background\\"><div class=\\"md-ink-container\\"></div>Cancel</button><button id=\\"launchmetricsDialogAcceptBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block md-btn--dialog btn--green btn--no-background\\"><div class=\\"md-ink-container\\"></div>Accept</button></footer></div></span>"`;
